### PR TITLE
feat(webui): change internals for shortcuts. add new shortcuts for settings in #172.

### PR DESCRIPTION
**Pull request by primetoxinz**
Rewrote how shortcuts are handled, having static actions and a way to change the key that triggers the action, allows for room to configure the keys in the future .

Add the new shortcuts as described in #172 along with a snackbar notification to indicate the change.

### CODE REVIEW

**Comment by primetoxinz** [comment]
May be nice to have a graphic showing a keyboard with labels to the action shortcut. It would be possible to do it dynamically rather than a static image, but it would make most sense to leave to if/when the keys themselves are configurable either per-client or server side

File: komga-webui/src/functions/shortcuts.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+// consider making this configurable on the server side?
+const keyMapping = {
+  'PageUp': Shortcut.SEEK_FORWARD,
+  'ArrowRight': Shortcut.SEEK_FORWARD,
+  'PageDown': Shortcut.SEEK_BACKWARD,
+  'ArrowLeft': Shortcut.SEEK_BACKWARD,
+  'ArrowDown': Shortcut.SEEK_UP,
+  'ArrowUp': Shortcut.SEEK_DOWN,
```
Comment (gotson) [inline komga-webui/src/functions/shortcuts.ts:42]: Those are inverted

File: komga-webui/src/functions/shortcuts.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    if (ctx.toolbar) {
+      ctx.toolbar = false
+      return
+    }
+    ctx.closeBook()
+  },
+} as Shortcuts
+
+export function executeShortcut (ctx: any, e: KeyboardEvent): boolean {
```
Comment (gotson) [inline komga-webui/src/functions/shortcuts.ts:92]: can we make `ctx` something else than `any` ?

Comment (primetoxinz) [reply komga-webui/src/functions/shortcuts.ts:92]: Unfortunately the Vue doesn't have a type, so I'm not sure what to make it

**Comment by gotson** [comment]
> May be nice to have a graphic showing a keyboard with labels to the action shortcut. It would be possible to do it dynamically rather than a static image, but it would make most sense to leave to if/when the keys themselves are configurable either per-client or server side

I was thinking indeed of removing the shortcuts from the documentation, and adding a _Help_ dialog in the reader. Maybe a button in the top toolbar, that would display a dialog showing the different shortcuts. We used to have this in the previous reader, we could reinstate it (in a different form).

File: komga-webui/src/functions/shortcuts.ts (hunk 2 above)
Comment (gotson) [reply komga-webui/src/functions/shortcuts.ts:92]: Indeed. Vue has type `Vue`, but the `BookReader` extension doesn't have any, so the functions are not recognized.

**Pull request merged**